measure.rb

```diff
@@ -1932,6 +1945,30 @@ def self.add_hot_water_and_appliances(runner, model, building, weather, spaces)
     return true
   end
 
+  def self.get_desuperheatercoil(hvac_map, relatedhvac, wh_id)
+    # search for the related cooling coil object for desuperheater
+
+    coil_found = false
+    # Supported cooling coil options
+    clg_coil_supported = [OpenStudio::Model::CoilCoolingDXSingleSpeed, OpenStudio::Model::CoilCoolingDXMultiSpeed, OpenStudio::Model::CoilCoolingWaterToAirHeatPumpEquationFit]
+    if hvac_map.keys.include? relatedhvac
+      target_hvac = hvac_map[relatedhvac]
+      target_hvac.each do |comp|
+        clg_coil_supported.each do |coiltype|
+          if comp.is_a? coiltype
+            return comp
+            coil_found = true
```

Comment:
Something is weird here. You're returning from the method right above assigning a value (which means the `coil_found = true` will never execute).

Reply:
If indeed the intent is to return with the coil as soon as you find it, you can probably remove the `coil_found` variable altogether. If you get to the code after the loop, by definition it will mean that the coil wasn't found.